pkg/tests/cross-tests/diff_cross_test.go

```diff
@@ -51,39 +51,191 @@ func TestUnchangedBasicObject(t *testing.T) {
 	})
 }
 
-func TestSimpleStringNoChange(t *testing.T) {
-	config := map[string]any{"name": "A"}
-	runDiffCheck(t, diffTestCase{
-		Resource: &schema.Resource{
-			Schema: map[string]*schema.Schema{
-				"name": {
-					Type:     schema.TypeString,
-					Optional: true,
-				},
+func TestDiffBasicTypes(t *testing.T) {
+	res := &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"other_prop": {
+				Type:     schema.TypeString,
+				Optional: true,
 			},
 		},
-		Config1: config,
-		Config2: config,
-	})
-}
+	}
 
-func TestSimpleStringRename(t *testing.T) {
-	runDiffCheck(t, diffTestCase{
-		Resource: &schema.Resource{
-			Schema: map[string]*schema.Schema{
-				"name": {
-					Type:     schema.TypeString,
-					Optional: true,
+	typeCases := []struct {
+		name             string
+		config1, config2 map[string]any
+		prop             *schema.Schema
+	}{
+		{
```

Comment:
The tests cases would be stronger if they've asserted not just TF=Pulumi conformance but also asserted what kind of plan ended up being picked. That'd be very nice if we could do that.